File: C++/lab2task2/lab2task2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");
    vector<int> nums;

    int n;
    cout << "Введите количество элементов: ";
    cin >> n;

    // Проверяем, что введенное количество элементов неотрицательное число
    if (n <= 0) {
        cout << "Ошибка: введено некорректное количество элементов." << endl;
        return 1; // Возвращаем ненулевой код, чтобы указать на ошибку
    }

    cout << "Введите числа: " << endl;;
    for (int i = 0; i < n; ++i) {
        int num;
        cin >> num;
        nums.push_back(num);
    }

    int j = 0;
    for (auto num : nums) {
        int digitCount = 0;
        int temp = num;

        // Подсчитываем количество цифр в числе
        while (temp != 0) {
            temp /= 10;
            digitCount++;
        }

        // Если количество цифр нечетное, увеличиваем счетчик
        if (digitCount % 2 != 0) {
            j++;
        }
    }

    cout << "Количество чисел с нечетным количеством цифр: " << j << endl;

    return 0;
}

File: Kotlin/lab2task2/lab2task2.kt
import java.util.Scanner

fun main() {
    // Создаем сканнер для ввода данных
    val scanner = Scanner(System.`in`)

    // Запрашиваем количество элементов у пользователя
    print("Введите количество элементов: ")
    val n = scanner.nextInt()

    // Создаем список для хранения чисел
    val nums = mutableListOf<Int>()

    // Запрашиваем числа у пользователя
    println("Введите числа:")
    repeat(n) {
        val num = scanner.nextInt()
        nums.add(num)
    }

    // Переменная для подсчета чисел с нечетным количеством цифр
    var count = 0

    // Проходим по каждому числу в списке
    for (num in nums) {
        var digitCount = 0
        var temp = num

        // Подсчитываем количество цифр в числе
        while (temp != 0) {
            temp /= 10
            digitCount++
        }

        // Если количество цифр нечетное, увеличиваем счетчик
        if (digitCount % 2 != 0) {
            count++
        }
    }

    // Выводим количество чисел с нечетным количеством цифр
    println("Количество чисел с нечетным количеством цифр: $count")
}
